src/lib/shadow-shim-helper-rs/src/shim_shmem.rs

```diff
@@ -402,27 +404,29 @@ impl SiginfoWrapper {
     }
 
     pub fn errno(&self) -> &libc::c_int {
-        &self.0.si_errno
+        self.0.errno()
     }
 
     pub fn errno_mut(&mut self) -> &mut libc::c_int {
-        &mut self.0.si_errno
+        self.0.errno_mut()
     }
 
     pub fn code(&self) -> &libc::c_int {
-        &self.0.si_code
+        self.0.code()
     }
 
     pub fn code_mut(&mut self) -> &mut libc::c_int {
-        &mut self.0.si_code
+        self.0.code_mut()
     }
 
     /// # Safety
     ///
     /// Pointer fields must not be dereferenced from outside of their
     /// native virtual address space.
     pub unsafe fn as_siginfo(&self) -> &libc::siginfo_t {
-        &self.0
+        static_assertions::assert_eq_align!(SiginfoWrapper, libc::siginfo_t);
+        static_assertions::assert_eq_size!(SiginfoWrapper, libc::siginfo_t);
+        unsafe { core::mem::transmute(self) }
     }
```

Comment:
Why is this function marked `unsafe`, but `linux_siginfo_t::as_bound_type` isn't?

Reply:
The `unsafe` here is meant to reflect that pointers may not be valid in the current address space. I'm fairly certain there aren't actually any pointers in `siginfo_t` though.

I ended up removing this wrapper altogether.